framework/src/outputs/FileOutput.C

```diff
@@ -64,7 +64,14 @@ FileOutput::FileOutput(const InputParameters & parameters)
     _file_num = 0;
 
   if (isParamValid("file_base"))
+  { // Check that we are the only process or not a subapp
+    if (!_app.isUltimateMaster())
+      if (_app.multiAppNumber() > 0)
+        mooseError("A file base has been specified but there are multiple child apps for this "
+                   "MultiApp. The file base would be obeyed and all child apps would write to the "
+                   "same file. This is disallowed.");
```

Comment:
```suggestion
        mooseError("The parameter 'file_base' may not be specified for a child app when the MultiApp has multiple instances of the child app.");
```

Reply:
Give the reason why as a comment - doesn't need to be in error message.

Reply:
Note this suggestion will require clang-formatting.

Reply:
why would we not tell the user why we disallow it

Reply:
IMO: it's not relevant to the user. All they need to know is: they can't do that. The technical reasons for why are for the developers. However, I don't feel strongly either way, so choose whichever you prefer.

Reply:
In general, I just try to not give users more information than they need, to avoid confusion/intimidation.

Reply:
I view it differently. If we re going to crash someone s simulation with an error message they need to know as much as possible. If they feel they have to dive into the code to understand we re only dev friendly not user friendly

Reply:
now what I have now might not be optimal still

Reply:
Updated suggestion